File: Empty/app/src/main/java/com/gmail/l2t45s7e9/empty/domain/factories/ViewModelDetailsFactory.java
package com.gmail.l2t45s7e9.empty.domain.factories;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import java.util.Map;
import javax.inject.Inject;
import javax.inject.Provider;

public class ViewModelDetailsFactory implements ViewModelProvider.Factory {

    private final Map<Class<? extends ViewModel>, Provider<ViewModel>> creators;
    private String id;
    private int color;

    @Inject
    public ViewModelDetailsFactory(Map<Class<? extends ViewModel>, Provider<ViewModel>> creators, String id, int color) {
        this.creators = creators;
        this.id = id;
        this.color = color;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) creators.get(modelClass).get();
    }
}

File: Empty/library/src/main/java/com/gmail/l2t45s7e9/library/domain/factories/ViewModelDetailsFactory.kt
package com.gmail.l2t45s7e9.library.domain.factories

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import javax.inject.Inject
import javax.inject.Provider

class ViewModelDetailsFactory @Inject constructor(
    private val creators: Map<Class<out ViewModel>, @JvmSuppressWildcards Provider<ViewModel>>
) : ViewModelProvider.Factory {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return creators[modelClass]?.get() as T
    }
}